hybrasyl/Game.cs

```diff
@@ -106,88 +173,46 @@ public static void Main(string[] args)
-
-                if (String.IsNullOrEmpty(serverIp))
-                    serverIp = "127.0.0.1";
-
-                if (String.IsNullOrEmpty(lobbyPort))
-                    lobbyPort = "2610";
-
-                if (String.IsNullOrEmpty(loginPort))
-                    loginPort = "2611";
-
-                if (String.IsNullOrEmpty(worldPort))
-                    worldPort = "2612";
-
-                Logger.InfoFormat("Using {0}: {1}, {2}, {3}", serverIp, lobbyPort, loginPort, worldPort);
-
-                Console.Write("Now, we will configure the Redis store.\n\n");
-                Console.Write("Redis IP or hostname (default is localhost): ");
-                var redisHost = Console.ReadLine();
-
-                Console.Write("Redis authentication information (optional - if you don't have these, just hit enter)");
-                Console.Write("Username: ");
-                var redisUser = Console.ReadLine();
-
-                Console.Write("Password: ");
-                var redisPass = Console.ReadLine();
-
-                if (String.IsNullOrEmpty(redisHost))
-                    redisHost = "localhost";
-
-                Config = new HybrasylConfig {datastore = {host = redisHost}, network =
+                var validConfig = false;
+                while (validConfig == false)
                 {
-                    lobby = new NetworkInfo { bindaddress = serverIp, port = Convert.ToUInt16(lobbyPort) },
-                    login = new NetworkInfo { bindaddress = serverIp, port = Convert.ToUInt16(loginPort) },
-                    world = new NetworkInfo { bindaddress = serverIp, port = Convert.ToUInt16(worldPort) }
-                }};
-
-                if (String.IsNullOrEmpty(redisUser))
-                    Config.datastore.username = redisUser;
-
-                if (String.IsNullOrEmpty(redisPass))
-                    Config.datastore.username = redisPass;
-
-                Config.SaveToFile(Path.Combine(Constants.DataDirectory, "config.xml"));
+                    try
+                    {
+                        Config = GatherConfig();
+                        validConfig = true;
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.ErrorFormat("Some of the values you entered were invalid. Try again. Error was: {0}",
+                            e.ToString());
+
+                    }
+                }
+                // Write out our configuration
+                XML.Serializer.Serialize(new XmlTextWriter(hybconfig, null), Config);
             }
-            ((log4net.Repository.Hierarchy.Hierarchy)LogManager.GetRepository()).Root.Level = Level.Debug;
-            ((log4net.Repository.Hierarchy.Hierarchy)LogManager.GetRepository()).RaiseConfigurationChanged(
+            // Set default logging level
+            ((log4net.Repository.Hierarchy.Hierarchy) LogManager.GetRepository()).Root.Level = Level.Info;

```

Comment:
this seems like a terribly redundant cast. Why not declare the using for Hierarchy, and only then need to cast as (Hierarchy)?